src/main/scala/io/iohk/ethereum/blockchain/sync/FastSync.scala

```diff
@@ -57,6 +57,10 @@ trait FastSync {
     case PeerActor.MessageReceived(BlockHeaders(Seq(blockHeader))) =>
       sender() ! PeerActor.Unsubscribe
 
+      if (appStateStorage.getEstimatedHighestBlock() < blockHeader.number) {
+        appStateStorage.putEstimatedHighestBlock(blockHeader.number)
+      }
```

Comment:
That's the question I've asked in the description. It would require us to additionally ask peers for their best blocks